File: 인터넷 강의/python_ruby/Class_member/2.py
class Cs:
    
    @staticmethod
    def static_method():
        print("Static method")
    
    @Classmethod
    def class_method(cls):
        print("Class method")

    def instance_method(self):
        print("Instance method")

i = Cs()

# ! python에서는 Class method가
# * static method
# * class method
# 두가지로 존재한다고 할 수 있다.

Cs.static_method()
Cs.class_method()
i.instance_method()

File: 인터넷 강의/python_ruby/Class_member/2.rb
class Cs
    #! Class method의 경우 Class Name이 반드시 기재되어야 한다.
    # ? instance 에서 class method를 호출하거나, Class에서 instance method를 호출한다면 에러가 발생한다.
    def Cs.class_method()
        p "Class method"
    end

    def instance_method()
        p "Instance method"
    end
end

i = Cs.new()
Cs.class_method()

i.instance_method()